index.js

```diff
@@ -1154,9 +1158,9 @@
 
       $pairs.push ([x, y]);
       try {
-        return Z.Setoid.test (x) &&
-               Z.Setoid.test (y) &&
-               Z.Setoid.methods.equals (x) (y);
+        return Z.Setoid.test (x) ?
+               Z.Setoid.methods.equals (x) (y) :
+               Object$prototype$equals.call (x, y);
```

Comment:
This line wrapping would be more consistent with the existing code:

```suggestion
        return Z.Setoid.test (x) ?
          Z.Setoid.methods.equals (x) (y) :
          Object$prototype$equals.call (x, y);
```